Tools/clinic/clinic.py

```diff
@@ -432,10 +432,11 @@ class FormatCounterFormatter(string.Formatter):
          the counts dict would now look like
          {'a': 2, 'b': 1, 'c': 1}
     """
-    def __init__(self):
-        self.counts = collections.Counter()
+    def __init__(self) -> None:
+        self.counts: dict[str, int] = collections.Counter()
 
-    def get_value(self, key, args, kwargs):
+    def get_value(self, key: str | int, args, kwargs) -> str:
+        assert isinstance(key, str)
```

Comment:
I would just do:

```suggestion
    def get_value(self, key: str, args, kwargs) -> str:  # type: ignore[override]
```

Mypy's complaining that we're overriding the superclass's method in a way that violates the Liskov Substitution Principle. But that's okay, we know what we're doing here; it's fine in this instance to just tell mypy to go away.